datafusion/physical-plan/src/tree_node.rs

```diff
@@ -35,3 +38,58 @@ impl DynTreeNode for dyn ExecutionPlan {
         with_new_children_if_necessary(arc_self, new_children).map(Transformed::into)
     }
 }
+
+#[derive(Debug)]
+pub struct PlanContext<T: Sized> {
```

Comment:
Could we also document what this is used for? It seems it is used for rewriting execution plans where some additional information needs to be stored in addition to the plan itself

Maybe a more descriptive name would be `PlanAndContext`? Though perhaps that is too verbose